cmd/thanos/compact.go

```diff
@@ -416,6 +441,23 @@ func runCompact(
 
 		srv.Handle("/", r)
 
+		// Periodically remove partial blocks and blocks marked for deletion
+		// since one iteration potentially could take a long time.
+		if conf.cleanupBlocksInterval > 0 {
+			g.Add(func() error {
+				// Wait the whole period at the beginning because we've executed this on boot.
```

Comment:
So... why not just removing this and removing boot time execution? (: Same stuff right?

Reply:
No, not really, because we want to give our users the ability to disable this concurrent cleaning for some reason i.e. to save up on S3 GET/LIST operation costs

Reply:
It sounds wrong that we do more complex code only because we don't want to change tests :thinking:

Reply:
The tests would be much more complex and probably out of the scope of this PR. Actually, it's not just that, I think it's nice that we do this at least once. Imagine where someone doesn't use `--wait` and the whole Thanos Compact process ended before the clean-up has happened. Space usage would never go down in the remote object storage even though it could. And the user then could be charged more as a result of this not happening.